### PR TITLE
Extract Python into an extension

### LINKED ISSUE #7296: Python lsp: Pyright doesn't good
### Check for existing issues

- [X] Completed

### Describe the feature

it should be pylsp. The pyright has issues about recognize libraries installed

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by PushUpek**
Partially resolves #7296

Release Notes:

- [x] support for python extracted to extension
- [x] using lsp servers installed by user
- [x] support for pyright
- [x] added support for basedpyright lsp server
- [x] defined tasks in tasks.json

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -401,7 +402,7 @@ tree-sitter-jsdoc = { git = "https://github.com/tree-sitter/tree-sitter-jsdoc",
 tree-sitter-json = "0.20.2"
 tree-sitter-markdown = { git = "https://github.com/MDeiml/tree-sitter-markdown", rev = "330ecab87a3e3a7211ac69bbadc19eabecdb1cca" }
 tree-sitter-proto = { git = "https://github.com/rewinfrey/tree-sitter-proto", rev = "36d54f288aee112f13a67b550ad32634d0c2cb52" }
-tree-sitter-python = "0.20.2"
+tree-sitter-python = "0.21.0"
```
Comment (maxdeviant) [inline Cargo.toml:405]: Do we still need this grammar as a dependency?

File: docs/src/languages/python.md [hunk 2]
```diff
@@ -1,10 +1,29 @@
 # Python
 
-Python support is available natively in Zed.
+Python support is available through the [Python extension](ttps://github.com/zed-industries/zed/tree/main/extensions/python).
```
Comment (maxdeviant) [inline docs/src/languages/python.md:3]: Broken link:

```suggestion
Python support is available through the [Python extension](https://github.com/zed-industries/zed/tree/main/extensions/python).
```

File: extensions/python/src/language_servers/commons.rs [hunk 3]
```diff
@@ -0,0 +1,92 @@
+use zed::lsp::{Completion, CompletionKind, Symbol, SymbolKind};
+use zed::{CodeLabel, CodeLabelSpan};
+use zed_extension_api::settings::LspSettings;
+use zed_extension_api::{self as zed, serde_json, Result};
+
+pub trait CommonPythonLSP {
```
Comment (maxdeviant) [inline extensions/python/src/language_servers/commons.rs:6]:
```suggestion
pub trait CommonPythonLsp {
```

File: Cargo.toml (hunk 1 above)
Comment (PushUpek) [reply Cargo.toml:405]: you are right, i should remove that dependency

**Comment by maxdeviant** [comment]
Thank you for putting this PR together, @PushUpek.

After some internal discussion, we've decided we want to leave the Python support implemented natively, for now.

There is a [PR](https://github.com/zed-industries/extensions/pull/1080) open to add `basedpyright` support as an extension, if that is something you are looking for.

**Comment by m1guer** [comment]
Thanks for the mention. The PR can be approved now; I fixed the CI/CD errors.

**Pull request closed without merging**